src/openms/source/ANALYSIS/ID/IDRipper.cpp

```diff
@@ -160,8 +187,84 @@ namespace OpenMS
         peptide_idents.push_back(*pep_it);
 
         //create and insert new map entry
-        ripped.insert(make_pair(file_, make_pair(protein_idents, peptide_idents)));
+        ripped.insert(make_pair(idx_pair, make_pair(protein_idents, peptide_idents)));
+      }
+    }
+    // Reduce the spectra data string list if that's what we ripped by
+    if (origin_annotation_fmt == MAP_INDEX || origin_annotation_fmt == ID_MERGE_INDEX)
+    {
+      map<pair<UInt, UInt>, pair<vector<ProteinIdentification>, vector<PeptideIdentification> > >::iterator it;
+      for (it = ripped.begin(); it != ripped.end(); ++it)
+      {
+        UInt const spectra_data_idx = it->first.second;
+        ProteinIdentification & prot_id = it->second.first.at(0);
+        StringList old_list;
+        prot_id.getPrimaryMSRunPath(old_list);
+        StringList new_list;
+        new_list.push_back(old_list.at(spectra_data_idx));
+        prot_id.setPrimaryMSRunPath(new_list);
+      }
+    }
+  }
+
+bool setOriginAnnotationMode_(short & mode, short const new_mode)
+{
+  if (mode != -1 && mode != new_mode)
+  {
+    return false;
+  }
+  mode = new_mode;
+  return true;
+}
+
+IDRipper::OriginAnnotationFormat IDRipper::detectOriginAnnotationFormat_(map<String, UInt> & file_origin_map, const std::vector<PeptideIdentification> & peptide_idents)
+  {
+    // In case we observe 'file_origin' meta values, we assign an index to every unique meta value
+    file_origin_map.clear();
+
+    vector<string> modes = { "file_origin", "map_index", "id_merge_index" };
+    short mode = -1;
```

Comment:
why not use `OriginAnnotationFormat mode = UNKNOWN`?
Would safe some `if ...`s below